### ISSUE TITLE
Media: add upload indicator % or progress bar

### ISSUE THREAD

**Issue by folletto** [Automattic/wp-calypso#12012]
Currently we just show a simple animation:

![cap-](https://cloud.githubusercontent.com/assets/4389/23800468/69d04340-05a4-11e7-864f-9e99f3fee672.gif)

This is ok for small files, but for files that are bigger or slower connections, not having any indication is a bit of a problem. Can be frustrating to not know how long does it take for something to upload.

We should thus add some kind of indicator, for example:

1. A % of progress
2. A visual progress bar
3. The time remaining
4. Else

Some initial options to be reviewed:

![loaders](https://cloud.githubusercontent.com/assets/4389/23800786/92fa445e-05a5-11e7-91b5-2522a4113eb2.png)

/cc @iamtakashi

**Comment by iamtakashi** [comment]
This is a good enhancement. To make this useful, we'd need to avoid thumbnails getting too small first. As you suggested [here](https://github.com/Automattic/wp-calypso/issues/9136#issuecomment-276165953), we should improve the sizing option.

I personally like Alt 1 or 2.

**Comment by folletto** [comment]
A couple more variations from ideas by @MichaelArestad: 

![media-loader](https://cloud.githubusercontent.com/assets/4389/23951562/4d8e2ab8-0986-11e7-9328-da6f8fef3b06.png)

**Comment by MichaelArestad** [comment]
I think after seeing it, alt 6 seems the nicest. Maybe lighten the gray a bit (or will it be a thumbnail of the uploaded image)?

**Comment by folletto** [comment]
Yeah should be the thumbnail, the above is just the default gray :)

**Comment by iamtakashi** [comment]
I like Alt 6 too. Let's do this!

**Comment by folletto** [comment]
Still an important UX change.

**Comment by joeboydston** [comment]
+1 for this change. A customer just wrote in with this feature request.

>When I upload a video through wordpress.com, there is no way to tell how much 
longer it is going to take. There should be a progress bar, or there should be 
some way to use a transfer protocol that gives me more information, such as 
sftp.

https://woothemes.zendesk.com/agent/tickets/925169

**Comment by folletto** [comment]
Still an important UX change.

**Comment by folletto** [comment]
Still an important UX change.

**Comment by formosattic** [comment]
Was also tracked on pKdGS-12Z-p2

**Comment by Greatdane** [comment]
Requested here; 3540064-zd-woothemes
